modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/SourceConfigurer.java

```diff
@@ -116,6 +122,14 @@ public Source build(ValueResolvingContext context) throws MuleException {
         } else {
           context = ValueResolvingContext.builder(initialiserEvent, expressionManager).build();
           Scheduler scheduler = (Scheduler) valueResolver.resolve(context);
+          if (isRestart) {
```

Comment:
I would move this logic to another class, like a factory that given a scheduler and the boolean returns another scheduler